From 7f791223a7e9d7be0686521b7ef57a864082974a Mon Sep 17 00:00:00 2001
From: Ng Zhi An <zhin@google.com>
Date: Wed, 16 May 2018 13:22:33 -0700
Subject: [PATCH 8/9] Add click handler for qsb on boot to keep it responsive

Bug: None
Test: reboot device, qsb is responsive, and launches search,
after widgets initialized, qsb remote view takes over click

Change-Id: I30c01ce1ff3da63cc0324f822d21901eb6a49c2d
---
 src/com/android/launcher3/qsb/QsbWidgetHostView.java | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/com/android/launcher3/qsb/QsbWidgetHostView.java b/src/com/android/launcher3/qsb/QsbWidgetHostView.java
index a8a41f664..7d8a4db0e 100644
--- a/src/com/android/launcher3/qsb/QsbWidgetHostView.java
+++ b/src/com/android/launcher3/qsb/QsbWidgetHostView.java
@@ -73,6 +73,18 @@ public class QsbWidgetHostView extends AppWidgetHostView {
         return getDefaultView(this);
     }
 
+    @Override
+    protected View getDefaultView() {
+        View v = super.getDefaultView();
+        v.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Launcher.getLauncher(getContext()).startSearch("", false, null, true);
+            }
+        });
+        return v;
+    }
+
     public static View getDefaultView(ViewGroup parent) {
         View v = LayoutInflater.from(parent.getContext())
                 .inflate(R.layout.qsb_default_view, parent, false);
-- 
2.17.1

